src/poetry/core/packages/package.py

```diff
@@ -117,6 +117,9 @@ def __init__(
 
         self._yanked = yanked
 
+        # Currently only used for testing purposes
```

Comment:
this is not true, per the change at line 360.

Reply:
Sorry I should have clarified, this is happening in CI for a real project of mine, not just in testing. The addition of `self._urls` is for the tests i wrote in `test_metadata.py`, since `url` is a `@property` and can't be set directly.

https://github.com/rommapp/romm/actions/runs/12623342037/job/35174039307#step:10:3085

Reply:
I would like to avoid adding production code that is only required for testing. Can you use a `PropertyMock` instead? See https://github.com/python-poetry/poetry/blob/5d8f8800b2cbb53da8057965663844bd1e04e455/tests/console/commands/test_check.py#L82-L86 for example.

Reply:
I think this is entirely redundant in the current version, the tests can manipulate the homepage and repository-url fields directly